gen-val/src/generation/src/NIST.CVP.ACVTS.Libraries.Generation/XECDH/v1_0/KeyVer/TestCaseExpectations/TestCaseExpectationProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using NIST.CVP.ACVTS.Libraries.Common.ExtensionMethods;
+using NIST.CVP.ACVTS.Libraries.Generation.Core;
+using NIST.CVP.ACVTS.Libraries.Oracle.Abstractions.DispositionTypes;
+
+namespace NIST.CVP.ACVTS.Libraries.Generation.XECDH.v1_0.KeyVer.TestCaseExpectations
+{
+    public class TestCaseExpectationProvider : ITestCaseExpectationProvider<XecdhKeyDisposition>
+    {
+        private readonly ConcurrentQueue<TestCaseExpectationReason> _expectationReasons;
+
+        public TestCaseExpectationProvider(bool isSample = false)
+        {
+            var expectationReasons = new List<TestCaseExpectationReason>();
+
+            int countForEachCase = (isSample ? 2 : 5);
```

Comment:
For fast running algorithms there is not much need to vary the number of test cases for isSample = true vs false. People generally like to see the same number of test cases if possible for true vs false.

Reply:
Resolved in https://github.com/usnistgov/ACVP-Server/pull/391/commits/7fe5b87057246768baa076f7076f9be13e725528